sycl/include/CL/sycl/handler.hpp

```diff
@@ -590,83 +603,62 @@ class handler {
   }
 
   // single_task version with a kernel represented as a lambda.
-  template <typename KernelName, typename KernelType>
+  template <typename KernelName = std::nullptr_t, typename KernelType>
```

Comment:
What if user write something like this:
```c++
handler.single_task<std::nullptr>(cl::sycl::range<1>{1}, [=](cl::sycl::id<1> i) {});
```

Reply:
Compiler will issue an error

Reply:
I'm not sure it's expected behavior.
I can't find any reason for a compiler error in SYCL spec.
Could you clarify why compiler should issue an error for this code? Is this a compiler bug?

Reply:
Should it be:
handler.single_task<std::nullptr_t>(cl::sycl::range<1>{1}, [=](cl::sycl::id<1> i) {});
Or:
handler.single_task<decltype(nullptr)>(cl::sycl::range<1>{1}, [=](cl::sycl::id<1> i) {});

As I think it's correct to error out with:
 handler.single_task<std::nullptr>(cl::sycl::range<1>{1}, [=](cl::sycl::id<1> i) {});

As std::nullptr is considered a value as far as I'm aware. I could be wrong though, I've not actually tried to invoke this example myself, but I tried it in compiler explorer: https://godbolt.org/z/xnaKhY

Reply:
I think you are right. I'll change the "marker type" from nullptr_t to something user can't use as kernel name - cl::sycl::detail::auto_kernel_name. In this case std::nullptr_t as kernel name will work.

Reply:
Still does not compile, but this is another problem - SemaSYCL.cpp needs a fix to handle std::nullptr_t (and maybe some other "special" types) differently when generating integration header. I'll file an issue.

Reply:
https://github.com/intel/llvm/issues/247

Reply:
> As I think it's correct to error out with:

Right. I think that was a typo - nullptr_t was meant. It still does not compile.